### PR TITLE
[1LP][RFR] Some Generic obj fixture move to global

### PR DESCRIPTION
**Pull request by digitronik**
## Purpose or Intent

- Generic object creation is with `rest` OR `ruby` and hectic task
- we repetition of fixtures nearly common so moving to global so anyone can use this for setup.
- I need this for increasing coverage for custom button over generic definition/class. 

- **Note: Just moving fixture not changing implementation** 

### PRT Run
{{pytest: cfme/tests/generic_objects/test_instances.py cfme/tests/automate/custom_button/test_service_objects.py -sqvv}}

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0d3cce4dab69feb72eea0ca343e8ff84578d042d

The *global* fixture **`gen_rest_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_text_display*
    - *unassigned_btn_setup*

The *global* fixture **`gen_definition`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *generic_object_button_group*
    - *generic_object_button*

The *global* fixture **`generic_object`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_tag_ui*
    - *test_generic_objects_tag_rest*

The *global* fixture **`add_generic_object_to_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *objects*
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_with_buttons_ui*

The *local* fixture **`objects`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_open_url_service_obj*

The *local* fixture **`unassigned_btn_setup`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit bc7dfe808b28a4f0ded75a18c736664ff678ffa0

The *global* fixture **`gen_rest_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_text_display*
    - *unassigned_btn_setup*

The *global* fixture **`gen_definition`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *generic_object_button_group*
    - *generic_object_button*

The *global* fixture **`generic_object`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_tag_ui*
    - *test_generic_objects_tag_rest*

The *global* fixture **`add_generic_object_to_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *objects*
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_with_buttons_ui*

The *local* fixture **`objects`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_open_url_service_obj*

The *local* fixture **`unassigned_btn_setup`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by digitronik** [comment]
PRT failures not due to fixtures.

File: cfme/fixtures/generic_object.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    rest_service = appliance.rest_api.collections.services.action.create(
+        name=fauxfactory.gen_numeric_string(16, start="gen_rest_serv", separator="-"), display=True
+    )
+    rest_service = rest_service[0]
+    yield rest_service
+    rest_service.action.delete()
+
+
+@pytest.fixture(scope="module")
+def gen_definition(appliance):
```
Comment (john-dupuy) [inline cfme/fixtures/generic_object.py]: Since you use `generic` below, I would expect this to be named `generic_definition`

**Comment by dajoRH** [comment]
I detected some fixture changes in commit c29706a103665bc770912350ed243359c395f534

The *global* fixture **`gen_rest_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_text_display*
    - *unassigned_btn_setup*

The *global* fixture **`generic_definition`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *generic_object_button_group*
    - *generic_object_button*

The *global* fixture **`generic_object`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_tag_ui*
    - *test_generic_objects_tag_rest*

The *global* fixture **`add_generic_object_to_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *objects*
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_with_buttons_ui*

The *local* fixture **`objects`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_open_url_service_obj*

The *local* fixture **`unassigned_btn_setup`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

File: cfme/fixtures/generic_object.py (hunk 1 above)
Comment (digitronik) [reply cfme/fixtures/generic_object.py]: done

File: cfme/fixtures/generic_object.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+@pytest.fixture(scope="module")
+def gen_rest_service(appliance):
+    """Simple service create with rest for generic object association"""
+
+    rest_service = appliance.rest_api.collections.services.action.create(
+        name=fauxfactory.gen_numeric_string(16, start="gen_rest_serv", separator="-"), display=True
+    )
+    rest_service = rest_service[0]
+    yield rest_service
+    rest_service.action.delete()
```
[the remark this answers is not in the document]
Comment (valaparthvi) [reply cfme/fixtures/generic_object.py]: **Optional** - 
```
 rest_service = appliance.rest_api.collections.services.action.create(
        name=fauxfactory.gen_numeric_string(16, start="gen_rest_serv", separator="-"), display=True
    )[0]
yield rest_service
```

**Comment by dajoRH** [comment]
I detected some fixture changes in commit a7dc99a71513daf7064f8bc7468358116a0e1f58

The *global* fixture **`gen_rest_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_text_display*
    - *unassigned_btn_setup*

The *global* fixture **`generic_definition`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *generic_object_button_group*
    - *generic_object_button*

The *global* fixture **`generic_object`** is used in the following files:
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_tag_ui*
    - *test_generic_objects_tag_rest*

The *global* fixture **`add_generic_object_to_service`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *objects*
- **cfme/tests/generic_objects/test_instances.py**
    - *test_generic_objects_with_buttons_ui*

The *local* fixture **`objects`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**
    - *test_custom_button_open_url_service_obj*

The *local* fixture **`unassigned_btn_setup`** is used in the following files:
- **cfme/tests/automate/custom_button/test_service_objects.py**

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**